sw/device/lib/crypto/api.h

```diff
@@ -0,0 +1,1378 @@
+ */
+typedef enum rsa_hash {
+  // SHA2-256 hashing mode for RSA.
+  kRsaHashSha256 = 0,
+  // SHA2-384 hashing mode for RSA.
+  kRsaHashSha384 = 1,
+  // SHA2-512 hashing mode for RSA.
+  kRsaHashSha512 = 2,
+  // SHA3-384 hashing mode for RSA.
+  kRsaHashSha3_384 = 3
+} rsa_hash_t;
+
+/**
+ * Enum to define named elliptic curves.
+ */
+typedef enum ecc_named_curve {
+  // ECC named curve NIST P256.
+  kEccNamedCurveNistP256 = 0,
+  // ECC named curve NIST P384.
+  kEccNamedCurveNistP384 = 1,
+  // ECC named curve brainpool P256r1.
+  kEccNamedCurveBrainpoolP256R1 = 2
+} ecc_named_curve_t;
+
+/**
+ * Struct to handle ECDSA result.
+ */
+typedef struct ecc_signature {
+  // Length of ECC signature R parameter, in bytes.
+  size_t signature_r_len;
+  // ECC signature R parameter.
+  uint32_t *signature_r;
+  // Length of ECC signature S parameter, in bytes.
+  size_t signature_s_len;
+  // ECC signature S parameter.
+  uint32_t *signature_s;
+} ecc_signature_t;
+
+/**
+ * Enum to define EdDSA mode for signature.
+ */
+typedef enum eddsa_sign_mode {
+  // Signature mode EdDSA.
+  kEddsaSignModeEdDSA = 0,
+  // Signature mode Hashed EdDSA.
+  kEddsaSignModeHashEdDSA = 1
+} eddsa_sign_mode_t;
+
+/**
+ * Struct to handle ECC public key type.
+ *
+ * Length of public key coordinates (x,y) follows len(prime_P).
+ */
+typedef struct ecc_public_key {
+  // ECC public key x coordinate.
+  uint32_t *pubKey_x;
+  // ECC public key y coordinate.
+  uint32_t *pubKey_y;
+} ecc_public_key_t;
+
+/**
+ * Struct to handle EdDSA public key type.
+ *
+ * Length of encoded public key for ed25519 is 256 bits (32 bytes).
+ */
+typedef struct eddsa_public_key {
+  // Public key for ed25519
+  uint32_t *pubKey;
+} eddsa_public_key_t;
+
+/**
+ * Struct for domain parameters for a generic ECC Weierstrass curve.
+ */
+typedef struct ecc_domain {
+  // Length of prime P.
+  size_t len_prime_P;
+  // Value of prime P.
+  uint32_t *prime_P;
+  // Length of coefficient A.
+  size_t len_A;
+  // Value of coefficient A.
+  uint32_t *A;
+  // Length of coefficient B.
+  size_t len_B;
+  // Value of coefficient B.
+  uint32_t *B;
+  // Value of basepoint G’s x coordinate.
+  uint32_t *base_point_G_x;
+  // Value of basepoint G’s y coordinate.
+  uint32_t *base_point_G_y;
+  // Length of prime order of group.
+  size_t len_prime_order_Q;
+  // Value of prime order of group.
+  uint32_t *prime_order_Q;
+  // Cofactor of the curve.
+  uint32_t cofactor;
+  // Checksum value of the parameters.
+  uint32_t checksum;
+} ecc_domain_t;
+
+/**
+ * Struct to handle crypto data buffer with pointer and length.
+ */
+typedef struct crypto_uint8_buf {
+  // Pointer to the data.
+  uint8_t *data;
+  // Length of the data in bytes.
+  size_t len;
+} crypto_uint8_buf_t;
+
+/**
+ * Performs the AES initialization operation.
+ *
+ * The `aes_mode`and `aes_operation` are selected and the `key`, `iv`
+ * are loaded into the register.
+ *
+ * @param key Pointer to the blinded key struct with key shares
```

Comment:
Would you mind making sure every sentence ends in a period?

Reply:
Even for parameter descriptions? I can't find that in the style guide, and when I searched for `@param` there I found an example that looks like the above, where most of the sentences end in periods but `@param` and `@return` descriptions don't.

Reply:
I went through just now and made sure everything else ends in a period, though.